snappy/click.go

```diff
@@ -481,6 +483,129 @@ func removePackageBinaries(m *snapYaml, baseDir string) error {
 	return nil
 }
 
+// this is a bit confusing, a localestring in xdg is:
+// "key=value" or "key[locale]=" with locale as:
+//  lang_COUNTRY@MODFIER (or a subset of this)
+var desktopFileI18nPattern = `(|\[[a-zA-Z_@]+\])`
+var validDesktopFileLines = []*regexp.Regexp{
+	// headers
+	regexp.MustCompile(`^\[Desktop Entry\]$`),
+	regexp.MustCompile(`^\[Desktop Action`),
+	// whitespace lines
+	regexp.MustCompile(`^\s*$`),
+	// lines with comments
+	regexp.MustCompile(`^\s*#`),
+	// https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s05.html
+	regexp.MustCompile(`^Type=`),
+	regexp.MustCompile(`^Version=`),
+	regexp.MustCompile(fmt.Sprintf(`^Name%s=`, desktopFileI18nPattern)),
+	regexp.MustCompile(fmt.Sprintf(`^GenericName%s=`, desktopFileI18nPattern)),
+	regexp.MustCompile(`^NoDisplay=`),
+	regexp.MustCompile(fmt.Sprintf(`^Comment%s=`, desktopFileI18nPattern)),
+	regexp.MustCompile(`^Icon=`),
+	regexp.MustCompile(`^Hidden=`),
+	regexp.MustCompile(`^OnlyShowIn=`),
+	regexp.MustCompile(`^NotShowIn=`),
+	regexp.MustCompile(`^Exec=`),
+	regexp.MustCompile(`^TryExec=`),
+	regexp.MustCompile(`^Terminal=`),
+	regexp.MustCompile(`^Actions=`),
+	regexp.MustCompile(`^MimeType=`),
+	regexp.MustCompile(`^Categories=`),
+	regexp.MustCompile(fmt.Sprintf(`^Keywords%s=`, desktopFileI18nPattern)),
+	regexp.MustCompile(`^StartupNotify=`),
+	regexp.MustCompile(`^StartupWMClass`),
+}
+
+func isValidDesktopFilePrefix(line string) bool {
+	for _, re := range validDesktopFileLines {
+		if re.MatchString(line) {
+			return true
+		}
+	}
+	return false
+}
+
+func validExecLine(m *snapYaml, line string) bool {
+	if !strings.HasPrefix(line, "Exec=") && !strings.HasPrefix(line, "TryExec=") {
```

Comment:
That's a bit surprising, and may be the source of future bugs as these are not actually valid Exec lines. It's easy to fix that by inverting the logic: rename the function to badExecLine, and invert the false<=>true logic (and the call site test). Then, it's true that something not being an Exec line is also not a bad Exec line.